File: Temperature.py
'''

F = (C * 9/5) + 32
C = (F - 32) * 5/9

30c => 86f


'''
def to_Fheit(celcius):
    result = celcius * (9/5) + 32
    return (int(result) if result.is_integer() else round(result, 2))


def to_Celcius(fheit):
    result = (fheit - 32) * 5/9
    return (int(result) if result.is_integer() else round(result, 2))


def main():
    while True:
        try:
            temper = input("Enter the temperature value you want to convert: ").strip()

            temp = float(temper)

            dest = input("Enter the temperature you want to convert to. \n'c' for Celcius, 'f' for Fahrenheit: ").strip().lower()

            

            if (dest != 'c'  and dest != "f") or len(dest) != 1:
                raise ValueError("Enter either 'f' or 'c' as temperature destination")
            
            if dest == 'c':
                result = to_Celcius(temp)
                sign = '℃'
                deste = "Celcius"
            else:
                result = to_Fheit(temp)
                sign = '℉'
                deste = "Fahrenheit"
                
                

            print(f"{temper} to {deste} is {result}{sign}")
            
            break

        except Exception as e:
            print(f"{e} Please try again\n")

if __name__ == "__main__":
    while True:
        main()
        repeat = input("Do you want to run the program again?  y to continue, anything else to quit: ").strip().lower()
        if repeat != "y":
            break

File: Temperature.cpp
/*
F = (C * 9/5) + 32
C = (F - 32) * 5/9

30c => 86f

*/

#include <iostream>
using namespace std;


char to_lower(char n);

void caller(){
    for (int i = 1; i< 300; i++)
        cout << i << (char) i << '\n';
}


double toFheit(double celcius);

double toCelcius(double fheit);

int main(){
    
    cout << "!============== Temperature Converter in C++ ===============!\n";
    
    double source;
    cout << "Enter the value you want to convert from: ";
    cin >> source;

    char dest;
    cout << "Enter the unit you want to convert to: ";
    cin >> dest;

    dest = to_lower(dest);

    if (!(dest == 'f' || dest == 'c'))
        cout << "Enter either 'f' or 'c' only, please try again!\n";
    else if (dest == 'c')
        cout << source << " to Celcius is " << toCelcius(source) <<  'C';
    else
        cout << source << " to Fahrenheit is " << toFheit(source) << 'F';
    
     
    return 0;
}
// ℃℉

char to_lower(char n){
    if (n > 64 && n < 92)
        return (n + 32);
    else
        return n;
}


double toFheit(double celcius){
    return (celcius * 9/5.0) + 32;
}

double toCelcius(double fheit){
    return (fheit - 32) * 5/9.0;
}

